lib/ansible/modules/cloud/amazon/aws_ses_identity_dkim.py

```diff
@@ -0,0 +1,172 @@
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+
+def get_identity_dkim_settings(module, client, identity):
+    try:
+        response = client.get_identity_dkim_attributes(Identities=[identity])
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg='Failed to retrieve identity DKIM settings for {identity}'.format(identity=identity))
+    dkim_attributes = response['DkimAttributes']
+
+    return dkim_attributes[identity]
+
+
+def ses_verify_dkim_domain(module, client, identity):
+    domain = identity.split('@')[-1]
+    try:
+        response = client.verify_domain_dkim(Domain=domain)
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg='Failed to start DKIM verification for {domain}.'.format(domain=domain))
+    dkim_tokens = response['DkimTokens']
+    return dkim_tokens
+
+
+def enable_identity_dkim_settings(module, client, identity):
+    try:
+        response = client.set_identity_dkim_enabled(DkimEnabled=True, Identity=identity)
+        return response
+    except (BotoCoreError) as e:
+        module.fail_json_aws(e, msg='Failed to enable DKIM for {identity}.'.format(identity=identity))
+    except (ClientError) as e:
+        pass
+        # The enable call will always return a ClientError, complaining about missing DNS record, but still changes setting to 'enabled'.
+        # So once DNS validation goes through it should work already, hence we ignore errors here.
+
+
+def disable_identity_dkim_settings(module, client, identity):
+    try:
+        response = client.set_identity_dkim_enabled(DkimEnabled=False, Identity=identity)
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg='Failed to disable DKIM settings for {identity}'.format(identity=identity))
+    return response
+
+
+def main():
+    module = AnsibleAWSModule(
+        argument_spec={
+            "identity": dict(required=True, type='str'),
+            "state": dict(default='enabled', choices=['enabled', 'disabled']),
+        },
+        supports_check_mode=True,
+    )
+
+    # SES APIs seem to have a much lower throttling threshold than most of the rest of the AWS APIs.
+    # Docs say 1 call per second. This shouldn't actually be a big problem for normal usage, but
+    # the ansible build runs multiple instances of the test in parallel that's caused throttling
+    # failures so apply a jittered backoff to call SES calls.
+    client = module.client('ses', retry_decorator=AWSRetry.jittered_backoff())
+
+    identity = module.params.get('identity')
+    state = module.params.get('state')
+    changed = False
+
+    if state == 'enabled':
+        ses_verify_dkim_domain(module, client, identity)
+        enable_identity_dkim_settings(module, client, identity)
+        # TODO: find a way to easily detect changes (updates to identity)
+        changed = True
```

Comment:
Can we adopt the same change checking strategy used in the other SES modules (e.g. https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/aws_ses_identity_policy.py#L113)

Basically:

* Call get_identity_dkim_settings BEFORE this point
* Check if the desired state is different from the current state, if so, set changed=True and call the relevant methods to make it so
* If chnaged is true, call get_identity_dkim_settings AGAIN below where you have it to get the updated settings to return

Doing it this way works well because:
* We don't call APIs to change things if there's no need
* The module is idempotent and correctly sets the changed state
* check_mode can be easily implemented by just skipping the mutation operations (see other review comments)

Reply:
Implemented now